File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/file.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import OpenAiFileDeleteResponse, OpenAiFileList, OpenAiFileObject, UploadFileRequest

class FileApi:
    """file API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def list_files(self, params: Optional[Dict[str, Any]] = None) -> OpenAiFileList:
        """List files"""
        return self._client.get(f"/ai/v3/files", params=params)

    def upload(self, body: Optional[UploadFileRequest] = None, params: Optional[Dict[str, Any]] = None) -> OpenAiFileObject:
        """Upload file"""
        return self._client.post(f"/ai/v3/files", data=body, params=params)

    def retrieve_file_content(self, file_id: str) -> str:
        """Retrieve file content"""
        return self._client.get(f"/ai/v3/files/{file_id}/content")

    def retrieve(self, file_id: str) -> OpenAiFileObject:
        """Retrieve file"""
        return self._client.get(f"/ai/v3/files/{file_id}")

    def delete_file(self, file_id: str) -> OpenAiFileDeleteResponse:
        """Delete file"""
        return self._client.delete(f"/ai/v3/files/{file_id}")

File: sdkwork-ai-sdk-flutter/lib/src/api/file.dart
import '../http/client.dart';
import '../models.dart';

class FileApi {
  final HttpClient _client;
  
  FileApi(this._client);

  /// List files
  Future<OpenAiFileList?> listFiles(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/files'), params: params);
    return response is OpenAiFileList ? response : null;
  }

  /// Upload file
  Future<OpenAiFileObject?> upload(UploadFileRequest? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/files'), body: body, params: params, contentType: 'multipart/form-data');
    return response is OpenAiFileObject ? response : null;
  }

  /// Retrieve file content
  Future<String?> retrieveFileContent(String file_id) async {
    final response = await _client.get(ApiPaths.aiPath('/files/${file_id}/content'));
    return response is String ? response : null;
  }

  /// Retrieve file
  Future<OpenAiFileObject?> retrieve(String file_id) async {
    final response = await _client.get(ApiPaths.aiPath('/files/${file_id}'));
    return response is OpenAiFileObject ? response : null;
  }

  /// Delete file
  Future<OpenAiFileDeleteResponse?> deleteFile(String file_id) async {
    final response = await _client.delete(ApiPaths.aiPath('/files/${file_id}'));
    return response is OpenAiFileDeleteResponse ? response : null;
  }
}
